### PR TITLE
auto generated page not found link errors fixed

### PR DESCRIPTION
**Pull request by Nayananga**
Problem: - There were some linking errors like "page not found <<from some page>> to <<some other page>>" when I run "yarn run check" after run "yarn build" to locally build camel-website repository in my pc.

Solution: -  To fix that I had to fix .adoc files in camel repository. Even I fixed those .adoc files when I run "./mvnw -Pfastinstall install" in camel root directory some changes revert back to error giving state. So I fixed those .adocs in "core/camel-core/src/main/docs/eips" and "camel/components" then ran "./mvnw -Pfastinstall install" to regenerate  doc files correctly

Output: - This reduced the  number of "page not found" errors when I run "yarn run check" after "yarn build" locally in camel-website repository

### CODE REVIEW

**Comment by davsclaus** [comment]
Ah darn this PR now have a few merge conflicts. Wonder if you could try to fix those and update the PR?

**Comment by zregvart** [comment]
@Nayananga this is awesome, thank you! Can you rebase (I usually run `git pull --rebase`) on the latest master and solve any conflicts.

**Comment by Nayananga** [comment]
Hello everyone, I think I resolved conflicts. Can you check now?

File: components/camel-aws-s3/src/main/docs/aws-s3-component.adoc [hunk 1]
```diff
@@ -87,7 +87,7 @@ with the following path and query parameters:
 [width="100%",cols="2,5,^1,2",options="header"]
 |===
 | Name | Description | Default | Type
-| *amazonS3Client* (common) | Reference to a com.amazonaws.services.s3.AmazonS3 in the link:registry.htmlRegistry. |  | AmazonS3
+| *amazonS3Client* (common) | Reference to a com.amazonaws.services.s3.AmazonS3 in the xref:registry.adocRegistry. |  | AmazonS3
```
Comment (oscerd) [inline components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: This seems to be wrong

Comment (Nayananga) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: even I change s3 configuration class as xref:registry.adoc[Registry] notice the brackets.It the .adoc generate just like that. Any solutions?

Comment (Nayananga) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: or should I keep it as it is?

Comment (oscerd) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: We can check later the way it is generated. It's just minor stuff

Comment (Nayananga) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: ok so what should I do now regarding this matter?

Comment (oscerd) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: For the moment leave it as is. We can re-work the docs

Comment (Nayananga) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: okay, Thank you. (going back to my other tasks)

Comment (davsclaus) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: I would just change the doc to say .. in the registry. And then dont have any link. This is how we do it in all the other Camel components.

Comment (Nayananga) [reply components/camel-aws-s3/src/main/docs/aws-s3-component.adoc]: Hello Mr@davsclaus can you take a look now?

**Pull request merged**